site/app/templates/polls/ViewPoll.twig

```diff
@@ -196,6 +243,39 @@
                 }
                 Plotly.relayout(container, update);
             });
+
+            {% if user_admin and poll.isOpen() %}
+            let graph_data = data;
+
+            function updateHistogram(updates) {
+                const histogram = document.getElementById('chartContainer');
+
+                for (option of Object.keys(updates)) {
+                    graph_data[0].y[responseIndexMap[option]] += updates[option];
+                }
+                
+                Plotly.restyle(histogram, { y: [graph_data[0].y] });
+            }
+
+            function initializeInstructorSocketClient() {
+                window.socketClient = new WebSocketClient();
+                 
+                window.socketClient.onmessage = (msg) => {
+                    switch (msg.type) {
+                        case "update_histogram":
+                            updateHistogram(msg.message);
+                            break;
+                        default:
+                            console.error("Unknown web socket message received:", msg);
+                            break;
+                    }
+                };
+
+                window.socketClient.open('polls-{{ poll.getId() }}-instructor');
+            }
```

Comment:
Let's move these functions to `site/public/js/polls.js` so they get linted.  The rest of the code can probably stay for now, although it all needs a bit of a refactor eventually.